ui/perfherder/alerts/AlertStatusCountdown.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faClock } from '@fortawesome/free-regular-svg-icons';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
+
+import SimpleTooltip from '../../shared/SimpleTooltip';
+import {
+  getStatus,
+  getTimeDifference,
+  getCountdownText,
+} from '../perf-helpers/helpers';
+import { weekdays } from '../perf-helpers/constants';
+
+export default class AlertStatusCountdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    this.isWeekend = false;
+    this.showCountdownToTriageIcon = true;
+  }
+
+  getDueDateCountdownsStatus() {
+    const { alertSummary } = this.props;
+
+    const now = new Date(Date.now());
+    const currentDay = now.getDay();
+    const triageDueDate = new Date(alertSummary.triage_due_date);
+    const bugDueDate = new Date(alertSummary.bug_due_date);
+
+    const timeToTriageDifference = getTimeDifference(now, triageDueDate);
+    const timeToBugDifference = getTimeDifference(now, bugDueDate);
+
+    const countdowns = {
+      triage: '',
+      bug: '',
+    };
+
+    const accessedDuringWeekend =
+      currentDay === weekdays.saturday || currentDay === weekdays.sunday;
+    if (accessedDuringWeekend) {
+      this.showCountdownToTriageIcon = false;
+      return countdowns;
+    }
+    if (alertSummary.first_triaged) {
+      countdowns.bug = getCountdownText(now, bugDueDate, timeToBugDifference);
+    } else {
+      countdowns.triage = getCountdownText(
+        now,
+        triageDueDate,
+        timeToTriageDifference,
+      );
+      countdowns.bug = getCountdownText(now, bugDueDate, timeToBugDifference);
+    }
+
+    return countdowns;
+  }
+
+  getCountdownClass(countdown) {
+    if (countdown === 'Overdue') {
+      return 'due-date-overdue';
+    }
+    if (countdown.endsWith('hours left')) {
+      return 'due-date-today';
+    }
+    return 'due-date-ok';
+  }
```

Comment:
Maybe we could have a constant for these classes:
```
const countdownClasses = {
    overdue: 'due-date-overdue',
    today: 'due-date-today',
    ok: 'due-date-ok',
    done: 'due-date-done',
}
```
`'due-date-done'` is used more than once.

What do you think?

Reply:
Isn't this an overkill? We have one value out of 4 that is used twice. Adding the constant is more code than logic.
I'm not doing push-back, I'm debating. I'm all in for as much as clean and readable code.